### PR TITLE
[MC-15] - Create Password Input Component

### LINKED ISSUE #15: [Feature] - Create Password Input Component
**Description**
I want a password input box to be created that will be used as the default for the entire site, following the prototype design.
The password box must have:
- Icon at the beginning of the password box
- Icon at the end of the password box
- Must show and hide when clicking the button

**Additional context**
image

**Acceptance Criteria**
- Separate CSS files
- The code must be responsive
- Build successfully
- lint successfully
- Use color in variables
- Show/hide password

### PR DESCRIPTION
**Pull request by pedroemanuel42**
# Feature/Issue Description
This PR create a text input box for password with the functionality to hide or show the text written through the icon (eye icon) on the side.

# Changes Summary
- Add new scss archive for input box password;
- Add new fonts archives from google fonts;
- Add a new dependency from fontawesome;
- Add a new service for password input;
- Add a test for password input service.

# How Has This Been Tested?

- To test the code, I ran the application, went to components and validated the operation on localhost. 

# Images (Opcional)
image
image

### CODE REVIEW

**Review by RodrigoSoaresNascimento — DISMISSED** [review]
is this sum function really necessary?

**Comment by tvps20** [comment]
The confirmation message pattern is not being respected. As the delivery date is near, the demand will be considered, but not in the next.
[MC-task number message commit]

**Review by tvps20 — DISMISSED** [review]

File: src/assets/scss/input-password.scss [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    box-shadow: 2px 2px 4px rgba(0, 0, 0, 0.1);
+    padding-left: 38px;
+}
+
+.password-input::placeholder {
+    color: $other-dark;
+}
+
+// Icons
+#padlock-icon {
```
Comment (feelsantiago) [inline src/assets/scss/input-password.scss]: Should not use Id for this kind of component. Either query by assossiations tags or classes.

File: src/assets/scss/input-password.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    color: $other-dark;
+    position: relative;
+    top: 44px;
+    left: 17px;
+}
+
+#eye-icon {
```
Comment (feelsantiago) [inline src/assets/scss/input-password.scss]: Same here.

File: src/pages/components/main.js [hunk 3]
```diff
@@ -1,14 +1,12 @@
 import { LayoutProviderService } from '../../services/layout-provider.service';
+import { PasswordInputService } from '../../services/password-input.service';
 
+import '../../assets/scss/input-password.scss';
```
Comment (feelsantiago) [inline src/pages/components/main.js]: It's better to import this inside components/style.scss

File: src/services/tests/password-input.service.test.js [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+            const resultIcon = $('#icon').attr('class');
+            const resultInput = $('#input-password');
+
+            expect(resultType).toBe('password');
+            expect(resultIcon).toBe('fa-solid fa-eye-slash');
+            expect(resultInput.is(':focus')).toBe(true);
+        });
+
+        it('Should check if div has query', () => {
+            const div = `
```
Comment (feelsantiago) [inline src/services/tests/password-input.service.test.js]: Don't need this mock

File: src/services/tests/password-input.service.test.js [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+            const resultType = $('#input-password').attr('type');
+            const resultIcon = $('#icon').attr('class');
+            const resultInput = $('#input-password');
+
+            expect(resultType).toBe('password');
+            expect(resultIcon).toBe('fa-solid fa-eye-slash');
+            expect(resultInput.is(':focus')).toBe(true);
+        });
+
+        it('Should check if div has query', () => {
```
Comment (feelsantiago) [inline src/services/tests/password-input.service.test.js]: This description is confusing.

File: src/services/tests/password-input.service.test.js [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+            `;
+
+            $(document.body).html(div);
+
+            expect(() => {
+                PasswordInputService.initFor('#pass');
+            }).toThrow('Failed to query input');
+        });
+
+        it('Should check if icon has query', () => {
```
Comment (feelsantiago) [inline src/services/tests/password-input.service.test.js]: Same here

**Review by feelsantiago — CHANGES_REQUESTED** [review]

File: src/services/tests/password-input.service.test.js (hunk 6 above)
Comment (pedroemanuel42) [reply src/services/tests/password-input.service.test.js:98]: Done all things, thanks for the advices!

**Comment by pedroemanuel42** [comment]
Noted, thanks for the tips! @tvps20

**Comment by pedroemanuel42** [comment]
All done, thanks for the advices! @feelsantiago

**Comment by pedroemanuel42** [comment]
This was a test function, it was removed by @feelsantiago. @RodrigoSoaresNascimento

File: src/services/tests/password-input.service.test.js [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+            const resultType = $('#input-password').attr('type');
+            const resultIcon = $('#icon').attr('class');
+            const resultInput = $('#input-password');
+
+            expect(resultType).toBe('password');
+            expect(resultIcon).toBe('fa-solid fa-eye-slash');
+            expect(resultInput.is(':focus')).toBe(true);
+        });
+
+        it('Should throw error failed on query div', () => {
```
Comment (feelsantiago) [inline src/services/tests/password-input.service.test.js]: The description should make sanse. Like: Should throw error if query fail or not find element.

File: src/services/tests/password-input.service.test.js [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+            expect(() => {
+                PasswordInputService.initFor('#paz');
+            }).toThrow('Failed to query div');
+        });
+
+        it('Should throw error failed on query input', () => {
```
Comment (feelsantiago) [inline src/services/tests/password-input.service.test.js]: Same here

File: src/services/tests/password-input.service.test.js [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+            `;
+
+            $(document.body).html(div);
+
+            expect(() => {
+                PasswordInputService.initFor('#pass');
+            }).toThrow('Failed to query input');
+        });
+
+        it('Should throw error failed on query icon', () => {
```
Comment (feelsantiago) [inline src/services/tests/password-input.service.test.js]: Same

**Review by feelsantiago — CHANGES_REQUESTED** [review]

**Comment by pedroemanuel42** [comment]
All done!

**Review by tvps20 — DISMISSED** [review]

**Review by feelsantiago — DISMISSED** [review]

**Review by feelsantiago — APPROVED** [review]

**Review by tvps20 — APPROVED** [review]

**Pull request merged**